frontend/src/metabase/dashboard/actions/data-fetching.js

```diff
@@ -419,32 +418,69 @@ export const fetchCardData = createThunkAction(
   },
 );
 
-export const fetchDashboardCardData = createThunkAction(
-  FETCH_DASHBOARD_CARD_DATA,
-  options => (dispatch, getState) => {
+export const fetchDashboardCardData =
+  ({ isRefreshing, ...options } = {}) =>
+  (dispatch, getState) => {
     const dashboard = getDashboardComplete(getState());
     const selectedTabId = getSelectedTabId(getState());
-    const dashcardIds = [];
-    const promises = getCurrentTabDashboardCards(dashboard, selectedTabId)
-      .filter(({ dashcard }) => !isVirtualDashCard(dashcard))
-      .map(({ card, dashcard }) => {
-        dashcardIds.push(dashcard.id);
-        return dispatch(fetchCardData(card, dashcard, options)).then(() => {
-          return dispatch(updateLoadingTitle());
-        });
+
+    const loadingIds = getLoadingDashCards(getState()).loadingIds;
+    const nonVirtualDashcards = getCurrentTabDashboardCards(
+      dashboard,
+      selectedTabId,
+    ).filter(({ dashcard }) => !isVirtualDashCard(dashcard));
+
+    let nonVirtualDashcardsToFetch = [];
+    if (isRefreshing) {
+      nonVirtualDashcardsToFetch = nonVirtualDashcards.filter(
+        ({ dashcard }) => {
+          return !loadingIds.includes(dashcard.id);
+        },
+      );
+      const newLoadingIds = nonVirtualDashcardsToFetch.map(({ dashcard }) => {
+        return dashcard.id;
       });
 
-    dispatch(setDocumentTitle(t`0/${promises.length} loaded`));
+      dispatch({
+        type: FETCH_DASHBOARD_CARD_DATA,
+        payload: {
+          currentTime: performance.now(),
+          loadingIds: loadingIds.concat(newLoadingIds),
```

Comment:
When auto-refreshing, we will fetch only dash cards that aren't loaded. So, `newLoadingIds` will not have the same id as in `loadingIds`

Consider the following scenario:
- The first refresh we load card 1 and 2
- card 1 finished, card 2 is still loading
- The second refresh we load only card 1 since card 2 is still loading

In this case `loadingIds` from redux would be `[1]` and `newLoadingIds` would be `[2]`, so we need to send `[1, 2]` in the action so we set loading IDs correctly.